custom/icds_reports/tests/test_disha_dump.py

```diff
@@ -20,14 +20,15 @@ class CountableList(list):
             def count(self, *args, **kwargs):
                 return len(self)
 
-        data = [['a'], ['b'], ['b\u0105z']]
+        data = [['a'], ['b'], ["d\xc3\xa9f"]]
         disha_get_rows_mock.return_value = CountableList(data)
 
         month = date(2018, 8, 1)
         state = 'Andhra pradesh'
         with TransientTempfile() as temp_path:
             dump = DishaDump(state, month)
-            dump._write_data_in_chunks(temp_path)
+            with open(temp_path, 'w+b') as f:
+                dump._write_data_in_chunks(f)
             with open(temp_path, 'r', encoding='utf-8') as f:
```

Comment:
just out of curiosity,
you write in bytes but read unicodes and are them comparing the data with non unicode `"d\xc3\xa9f"`, that works?

Reply:
The data is written is utf8 encoded bytes, so when we read back, we specify the same encoding to read back those bytes, so it works.

Reply:
so f.read() gives you back 'utf8 encoded' data but we compare it with "d\xc3\xa9f" and that comes to be equal?

Reply:
Yeah. Because `d\xc3\xa9f` is also utf8 encoded string.